src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -286,7 +282,13 @@ class InstanceOpFactoryImpl(
           config.mesosBridgeName())
 
         val instanceId = reservedInstance.instanceId
-        val taskIds = reservedInstance.tasksMap.keys
+        val taskIds = if (reservedInstance.tasksMap.nonEmpty) {
```

Comment:
Why is there no check for `reservedInstance.hasReservation` as done for apps?

Reply:
I don't see where this is checked for apps, so maybe the comment is outdated?
If the instance has no reservation there could be no match I guess.